packages/asset-service/src/utils.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  CHAIN_NAMESPACE,
+  CHAIN_REFERENCE,
+  ChainId,
+  ChainNamespace,
+  ChainReference,
+  fromChainId
+} from '@shapeshiftoss/caip'
+
+// https://www.coingecko.com/en/api/documentation - See asset_platforms
+export const chainPartsToCoingeckoAssetPlatform = (
+  chainNamespace: ChainNamespace,
+  chainReference?: ChainReference
+): string => {
+  return (() => {
+    switch (chainNamespace) {
+      case CHAIN_NAMESPACE.Ethereum:
+        return 'ethereum'
+      case CHAIN_NAMESPACE.Cosmos:
+        return chainReference === CHAIN_REFERENCE.CosmosHubMainnet ? 'cosmos' : 'osmosis'
+      default:
+        throw new Error(
+          `chainNamespace ${chainNamespace}, chainReference ${chainReference} not supported.`
+        )
+    }
+  })()
+}
+
+export const chainIdToCoingeckoAssetPlatform = (chainId: ChainId): string => {
```

Comment:
Thoughts about just switching on `chainId` and cases being `KnownChainIds`? Seems like breaking it into parts and then having an optional `chainReference` just complicates things and if no `chainReference` is provided for `Cosmos` network it will choose `osmosis` which doesn't seem ideal. In regards to @gomesalexandre comment about bitcoin, we can also just add the bitcoin case and allow it to fall through to the default to be exhaustive (versus comment) or both I guess.

Reply:
This won't work if we want to support `ChainId`s other than mainnet for each `ChainNamespace`. If we don't need to support them that yes, this can be far more simple using `KnownChainIds`, but I'm not sure if that's what we want?

Reply:
https://github.com/shapeshift/lib/pull/793/commits/b5b6642c3a262d7b859ed6a8e2ef7b9ec5aa4723

- Removed "parts" function
- Fixed cosmos logic where it might chose osmosis